webapp/services.py

```diff
@@ -36,9 +47,16 @@ def find_assets(self, file_type="%", tag=None, query=None):
         if tag:
             tag_condition = Asset.tags.any(Tag.name == tag)
             conditions.append(tag_condition)
-        assets = db_session.query(Asset).filter(*conditions).yield_per(100)
-        for asset in assets:
-            yield asset
+        assets = (
+            db_session.query(Asset)
+            .filter(*conditions)
+            .order_by(order_by.desc() if desc_order else order_by)
+            .offset((max(page, 1) - 1) * max(per_page, 10))
```

Comment:
Why a maximum here, isn't per_page always defined to be 20 when called, even so why a max between 10 when by default we call it with 20